rcgen/src/lib.rs

```diff
@@ -1465,18 +1467,23 @@ fn write_distinguished_name(writer: DERWriter, dn: &DistinguishedName) {
 					match content {
 						DnValue::BmpString(s) => writer
 							.next()
-							.write_tagged_implicit(TAG_BMPSTRING, |writer| writer.write_bytes(s)),
-						DnValue::Ia5String(s) => writer.next().write_ia5_string(s),
-						DnValue::PrintableString(s) => writer.next().write_printable_string(s),
+							.write_tagged_implicit(TAG_BMPSTRING, |writer| {
```

Comment:
I think this could be switched to use Yasna's [`write_bmp_string`](https://qnighy.github.io/yasna.rs/yasna/struct.DERWriter.html#method.write_bmp_string) method to avoid managing the tag ourselves and to match the IA5 String and PrintableString match arms.

Reply:
As the inner type is a `Vec<u8>` encoded as UTF-16 BE it will force us to allocate a new string to be able to call this. 

One solution is to change the inner type to a `String`  but if one day we move from `yasna` we don' t have the guarantee that the new one will provide a helper to handle the UTF-8 to UTF-16 conversion.

Besides there is also no `write_universal_string` so for the `UniversalString` type, it makes no sense to have a `String` as an inner type for this one. So I would prefer keeping a `Vec<u8>` representation for both ` BmpString` and `UniversalString`

Reply:
> As the inner type is a Vec<u8> encoded as UTF-16 BE it will force us to allocate a new string to be able to call this.

Good point :+1: 

> So I would prefer keeping a Vec<u8> representation for both  BmpString and UniversalString

That sounds reasonable to me. Thanks for considering the feedback.